src/coreclr/jit/valuenum.cpp

```diff
@@ -8773,9 +8773,18 @@ bool Compiler::fgValueNumberConstLoad(GenTreeIndir* tree)
     if ((varTypeIsIntegral(tree) || varTypeIsFloating(tree)) &&
         GetStaticFieldSeqAndAddress(vnStore, tree->gtGetOp1(), &byteOffset, &fieldSeq))
     {
+#if defined(TARGET_XARCH) && defined(FEATURE_SIMD)
+        // SSE2 and AdvSimd are baselines so TYP_SIMD8-16 are always there
+        // for TYP_SIMD32 we need to check AVX support on XARCH
+        if (tree->TypeIs(TYP_SIMD32) && !compOpportunisticallyDependsOn(InstructionSet_AVX))
```

Comment:
How would we get a `TYP_SIMD32` tree here if we didn't have AVX? Seems like that would be a bug.

Reply:
Good point, removed